creator-node/src/snapbackSM.js

```diff
@@ -47,20 +51,175 @@ class SnapbackSM {
     )
   }
 
+  // Helper function to retrieve all config based
+  async getSPInfo () {
+    const spID = config.get('spID')
+    const endpoint = config.get('creatorNodeEndpoint')
+    const delegateOwnerWallet = config.get('delegateOwnerWallet')
+    const delegatePrivateKey = config.get('delegatePrivateKey')
+    return {
+      spID,
+      endpoint,
+      delegateOwnerWallet,
+      delegatePrivateKey
+    }
+  }
+
+  // Query eth-contracts chain for endpoint to ID info
+  async recoverSpID () {
+    if (config.get('spID') !== 0) {
+      this.log(`Known spID=${config.get('spID')}`)
+      return
+    }
+
+    const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
+      config.get('creatorNodeEndpoint')
+    )
+    this.log(`Recovered ${recoveredSpID} for ${config.get('creatorNodeEndpoint')}`)
+    config.set('spID', recoveredSpID)
+  }
+
+  /*
+    http://audius-disc-prov_web-server_1:5000/users/creator_node?creator_node_endpoint=http://cn2_creator-node_1:4001
+    Send request to discovery provider for all users with this node as primary
```

Comment:
limit maximum number of parallel syncs issued by adding a sync_status monitoring step to the syncQueue processing AND limiting bullQueue concurrency